### PR TITLE
Make vendor

### PR DESCRIPTION
**Pull request by m1kola**
### Which issue this PR addresses:

Updates vendor and fixes missing `sigs.k8s.io/yaml` from `go.mod`.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -100,6 +100,7 @@ require (
 	sigs.k8s.io/cluster-api-provider-openstack v0.3.2 // indirect
 	sigs.k8s.io/controller-runtime v0.6.3
 	sigs.k8s.io/controller-tools v0.3.0
```
Comment (mjudeikis) [inline go.mod:102]: run `make vendor`

Comment (m1kola) [reply go.mod:102]: Yep, just pushed it a second ago.

Comment (m1kola) [reply go.mod:102]: Hang on. I did `go mod vendor`. `make vendor` does something else. Let me double check.

Comment (m1kola) [reply go.mod:103]: @mjudeikis so `make vendor` updates vendor. I re-porpused this PR to update vendor (had to fix some incompatibility with `azure-sdk-for-go v49.0.0`) and submitted a separate PR just for docs: https://github.com/Azure/ARO-RP/pull/1214

Not sure if we actually want to update dependencies or just `go mod tidy` and `go mod vendor` to fix missing yaml package.

**Review by jim-minter — APPROVED** [review]
I don't quite understand why we're doing this, but I'm OK with doing it.

**Pull request merged**